src/chain/payment_record.cpp

```diff
@@ -322,5 +322,17 @@ void payment_record::set_hash(hash_digest&& hash)
     hash_ = std::move(hash);
 }
 
+uint32_t payment_record::index() const
+{
+    return index_;
+}
+
+void payment_record::set_index(uint32_t value)
+{
+    // This is no longer a default instance, so valid.
+    valid_ = true;
```

Comment:
Need to apply this to `payment_record::set_hash(hash_digest&& hash)` as well for consistency.

Reply:
Noted, in following commit.